private/buf/bufwire/image_reader.go

```diff
@@ -130,20 +131,32 @@ func (i *imageReader) GetImage(
 	if err != nil {
 		return nil, err
 	}
-	if len(externalDirOrFilePaths) == 0 {
+	if len(externalDirOrFilePaths) == 0 && len(excludeDirOrFilePaths) == 0 {
 		return image, nil
 	}
 	imagePaths := make([]string, len(externalDirOrFilePaths))
-	for i, externalDirOrFilePath := range externalDirOrFilePaths {
-		imagePath, err := imageRef.PathForExternalPath(externalDirOrFilePath)
-		if err != nil {
-			return nil, err
+	if len(externalDirOrFilePaths) > 0 {
```

Comment:
Can we drop this condition? We won't `range` over anything if the length is not greater than zero. Same thing several lines below.